utest/model/test_tags.py

```diff
@@ -144,6 +144,41 @@ def _verify_slice(self, sliced, expected):
         assert_true(isinstance(sliced, Tags))
         assert_equal(list(sliced), expected)
 
+    def test__eq__(self):
+        tags_a = [
+            'component.selenium', 'LAME_CI', 'boing.bom.bin',
+            'boing.bam.boom.baaaaaaaaaa',
+            'boing.bam.boom.baaaaaaaaaa.bla',
+            'boing.bing.bla', 'boing.bing.bleu', 'X_FOR_Y']
+
+        tags_b = [
+            'component',
+            'boing.component',
+            'boing',
+            'boing.bam.boom.bba',
+            'boing.bam.boom',
+            'boing.bam',
+            'boing.bing'
+        ]
+
+        assert_equal(Tags(['x']), Tags(['x']))
+        assert_equal(Tags(['X']), Tags(['x']))
+
+        assert_equal(Tags(tags_a + tags_b), Tags(tags_a + tags_b))
+        assert_equal(Tags(tags_a + tags_b), Tags(tags_a) + Tags(tags_b))
+        assert_equal(Tags(tags_b + tags_a), Tags(tags_a) + Tags(tags_b))
+        assert_equal(Tags(tags_a + tags_b), Tags(tags_b) + Tags(tags_a))
+        assert_equal(Tags(tags_b + tags_a), Tags(tags_b) + Tags(tags_a))
```

Comment:
I am trying to validate that adding the same tags, in any order, will still produce equivalent tag sets (e.g. no duplicates or sorting discrepancies)

Reply:
That's not related to equality though. There are also already some tests related to it. If needed, they can be enhanced.

Reply:
At least the existing `test_init_with_iterable_and_normalization_and_sorting` tests this already. If needed, it could be enhanced. It probably should also be split to at least two tests (`test_init_with_iterable` and `test_normalization_and_sorting`).

Reply:
You are probably right. I will add one more case to `test_init_with_iterable_and_normalization_and_sorting`, and remove most of this.

Reply:
Feel free to split that test to two (or even three) if it makes sense. I don't generally like the idea of one test testing various features but here it has removed duplication and the test itself is pretty short and easy to understand. If more cases are added, it probably is better to split it to separate explicit tests.